### PR TITLE
Adding branched from node. [ENG-1898]

### PR DESCRIPTION
**Pull request by corbinSanders**
## Purpose

To make analytics easier, a database column `branched_from_node` needs to be added.

## Changes

Adding field in model, adding code for population at registration time. Adding migration and management command to populate new field.

## QA Notes

Workflows impacted: creating registrations both ways

Does this change require a data migration? Yes

What is the level of risk?

Any permissions code touched?

Is this an additive or subtractive change, other?

How will QA verify? (UI, API, admin app or AdminAdminApp?) DevTest, Courtney directly queries DB

How will this impact performance?

## Documentation

N/A

## Side Effects

N/A

## Ticket

https://openscience.atlassian.net/browse/ENG-1898

### CODE REVIEW

File: osf/management/commands/populate_branched_from_node.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            '--page_size',
+            type=int,
+            default=10000,
+            help='How many rows to process at a time',
+        )
+
+    def handle(self, *args, **options):
+        script_start_time = datetime.datetime.now()
+        logger.info('Script started time: {}'.format(script_start_time))
+        logger.debug(options)
```
Comment (UdayVarkhedkar) [inline osf/management/commands/populate_branched_from_node.py:49]: Is this leftover from local development?

**Comment by UdayVarkhedkar** [comment]
Am I understanding correctly that the celery beat is necessary because the script will process 10,000 rows at a time so it doesn't take up too much time?

File: osf/management/commands/populate_branched_from_node.py (hunk 1 above)
Comment (corbinSanders) [reply osf/management/commands/populate_branched_from_node.py:49]: The logging? No, I want those logs to stay.

Comment (UdayVarkhedkar) [reply osf/management/commands/populate_branched_from_node.py:49]: Specifically L49 logging out the options dictionary - I don't have an issue with it but it's something that stuck out as something that could be leftover from debugging.

**Comment by corbinSanders** [comment]
> Am I understanding correctly that the celery beat is necessary because the script will process 10,000 rows at a time so it doesn't take up too much time?

Yes. I don't want to process all 80k registrations at once. This way those registrations will be processed over a week or so.

File: osf/management/commands/populate_branched_from_node.py (hunk 1 above)
Comment (corbinSanders) [reply osf/management/commands/populate_branched_from_node.py:49]: `logger.debug` wont display a log message unless the logging config level is set to DEBUG. Debug statements are common in our management commands.

File: osf/management/commands/populate_branched_from_node.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+logger = logging.getLogger(__name__)
+
+POPULATE_BRANCHED_FROM_NODE = """UPDATE osf_abstractnode a
+ SET branched_from_node = CASE WHEN
+    EXISTS(SELECT id FROM osf_nodelog WHERE action='project_created_from_draft_reg' AND node_id = a.id) THEN False
+    ELSE True
+END
+WHERE
+a.type = 'osf.registration' AND
+a.branched_from_node IS null
```
Comment (brianjgeiger) [inline osf/management/commands/populate_branched_from_node.py:17]: Do you need a LIMIT statement inside the sql in order to properly use the `page_size` parameter that you're passing in on L24? Something like `LIMIT %s`.

File: osf/management/commands/populate_branched_from_node.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+"""
+
+@celery_app.task(name='management.commands.populate_branched_from')
+def populate_branched_from(page_size, dry_run):
```
Comment (brianjgeiger) [inline osf/management/commands/populate_branched_from_node.py:21]: Does this need default parameters for running as a celery task? I didn't see any parameters being passed in from your crontab configuration.

File: osf/management/commands/populate_branched_from_node.py (hunk 2 above)
Comment (corbinSanders) [reply osf/management/commands/populate_branched_from_node.py]: Ah! Thanks!

File: osf/management/commands/populate_branched_from_node.py (hunk 3 above)
Comment (corbinSanders) [reply osf/management/commands/populate_branched_from_node.py]: Good catch!

**Pull request merged**